src/main/java/me/moodcat/backend/chat/ChatBackend.java

```diff
@@ -0,0 +1,145 @@
+package me.moodcat.backend.chat;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+import com.google.inject.Singleton;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import me.moodcat.database.controllers.ChatDAO;
+import me.moodcat.database.entities.ChatMessage;
+import me.moodcat.database.entities.Song;
+import org.eclipse.jetty.util.component.AbstractLifeCycle.AbstractLifeCycleListener;
+import org.eclipse.jetty.util.component.LifeCycle;
+
+import me.moodcat.database.entities.Room;
+import me.moodcat.database.controllers.RoomDAO;
+import me.moodcat.util.CallableInUnitOfWork.CallableInUnitOfWorkFactory;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * @author Jan-Willem Gmelig Meyling
+ */
+@Slf4j
+@Singleton
+public class ChatBackend extends AbstractLifeCycleListener {
+
+    private final ScheduledExecutorService executorService;
+    private final Map<Integer, ChatRoomInstance> roomInstances;
+    private final Provider<RoomDAO> roomDAOProvider;
+    private final Provider<ChatDAO> chatDAOProvider;
+    private final CallableInUnitOfWorkFactory callableInUnitOfWorkFactory;
+
+    @Inject
+    public ChatBackend(Provider<RoomDAO> roomDAOProvider,
+                       CallableInUnitOfWorkFactory callableInUnitOfWorkFactory,
+                       Provider<ChatDAO> chatDAOProvider) {
+        this.executorService = Executors.newScheduledThreadPool(4);
+        this.roomInstances = Maps.newHashMap();
+        this.roomDAOProvider = roomDAOProvider;
+        this.callableInUnitOfWorkFactory = callableInUnitOfWorkFactory;
+        this.chatDAOProvider = chatDAOProvider;
+    }
+
+    public void initializeInitialRooms() {
+        performInUnitOfWork(() -> {
+            RoomDAO roomDAO = roomDAOProvider.get();
+            roomDAO.listRooms().stream()
+                .map(ChatRoomInstance::new)
+                .forEach(room -> roomInstances.put(room.getRoom().getId(), room));
+            return null;
```

Comment:
The method returns a void and then the rooms are created asynchronously. Therefore the null pointer isn't accessible anywhere. In fact, runnables are always callables that return `null`. The alternative is to capture the stream of generated rooms, return this, and promise this as `Future`. But I don't think we ever need the instances of the created room instances.